core/domain/summary_services.py

```diff
@@ -163,6 +163,46 @@ def get_displayable_collection_summary_dicts_matching_ids(collection_ids):
     return _get_displayable_collection_summary_dicts(collection_summaries)
 
 
+def get_collection_node_exploration_dicts(
```

Comment:
Looking at this, the backend flow doesn't seem to have anything to do with collections -- it's more like "search for explorations and return their metadata dicts". The handler can still stay in collection_editor.py, I think, but can we rename this to get_exploration_metadata_dicts and the handler to ExplorationMetadataSearchHandler?